mongosql/src/air/desugarer/testdata/desugar_accumulators.yml

```diff
@@ -69,43 +69,176 @@ tests:
           },
       }
     expected:
-      - { "$group": { "_id": null, "acc": { "$addToSet": "$a" } } }
+      - {
+        "$group":
+          {
+            "_id": null,
+            "acc":
+              {
+                "$addToSet":
+                  {
+                    "$cond":
+                      [
+                        { "$in": [{ "$type": "$a" }, ["missing", "null"]] },
+                        "$$REMOVE",
+                        "$a",
+                      ]
+                  }
+              }
+          }
+      }
       - {
         "$project":
           {
             "_id": 1,
+            "acc": { "$size": "$acc" },
+          },
+      }
+
+  - name: "desugar non-distinct $sqlCount multi-column arg"
```

Comment:
Adding these 2 skipped tests really helped me understand the subtleties of this problem. It also helped to reveal the issue I discussed offline with Oliver re: `*` should be unconditional, and non-`*` should be conditional. Since I wrote the tests for understanding, I figured it was best just to leave them in with a `skip_reason`.